jscomp/others/js_option.ml

```diff
@@ -59,5 +59,27 @@ let andThen f x =
   | None -> None 
   | Some x -> f x [@bs]
 
+let map f x =
+  match x with
+  | None -> None
+  | Some x -> Some (f x [@bs])
```

Comment:
re: implementing both, I'd rather just have `flatMap` then. Having to memorize two functions is not in any way friendlier to newcomers than having to memorize just one. It's a non-solution that ignores the goal entirely.